tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -215,48 +218,63 @@ sub prepare_guest {
     my $vm = shift;
 
     assert_script_run "virsh dumpxml $vm > $vm.xml";
-    script_run "virsh destroy $vm";
-    assert_script_run "virsh undefine $vm";
-
-    #extra process for XEN hypervisor
-    if (is_xen_host) {
-
-        #enable pci-passthrough and set model for pv guest.
-        #refer to bug #1167217 for the reason
-        my $passthru_xml = "<passthrough state='on'/>";
-        my $e820_xml     = "";
-        if (is_pv_guest($vm)) {
-            $e820_xml = "<e820_host state='on'/>";
-        }
-        if (script_run("grep '<features>' $vm.xml") == 0) {
-            assert_script_run "sed -i \"/<features>/a\\<xen>\\n$passthru_xml\\n$e820_xml\\n</xen>\" $vm.xml";
-        }
-        else {
-            assert_script_run "sed -i \"/<domain /a\\<features>\\n<xen>\\n$passthru_xml\\n$e820_xml\\n</xen>\\n</features>\" $vm.xml";
-        }
-
-        #disable memory ballooning for fv guest as it is not supported
-        if (is_fv_guest($vm)) {
-            assert_script_run "sed -i '/<currentMemory/d' $vm.xml";
+    my $mem          = "";
+    my $cur_mem      = "";
+    my $is_ballooned = 0;
+    $mem          = script_output "xmlstarlet sel -t -v //memory $vm.xml";
+    $cur_mem      = script_output "xmlstarlet sel -t -v //currentMemory $vm.xml";
+    $is_ballooned = 1 if $cur_mem != $mem;
+
+    unless (is_sle('=12-SP5') && (is_kvm_host || (is_fv_guest($vm) && !$is_ballooned))) {
```

Comment:
Make sure your is_sle('=12-SP5') works on virt_autotest host

Reply:
It works in my 3 verification runs. What are you concern about this function? the usage or the actual sle versions in the host of virtualization tests with non-developing host?

Reply:
Yes.

Reply:
is_sle() works for checking host release in virtualization test as it is based on VERSION_TO_INSTALL/VERSION. it should not apply to guest or host upgrade test.

Reply:
If you want to check the version of guest system you may use `version_utils::get_os_release()` or `version_utils::check_os_release()`.